typescript/infra/src/core/govern.ts

```diff
@@ -42,39 +49,91 @@ export class AbacusCoreGovernor<Chain extends ChainName> {
   }
 
   logCalls() {
-    objMap(this.calls, (chain, calls) => {
-      console.log(chain, calls);
+    const logFn = async (
+      chain: Chain,
+      _: ChainConnection,
+      calls: types.CallData[],
+    ) => console.log(chain, calls);
+    return this.executeCalls(this.connectionFn, logFn);
+  }
+
+  protected async executeCalls(
+    connectionFn: (chain: Chain) => ChainConnection,
+    executeFn: (
+      chain: Chain,
+      connection: ChainConnection,
+      calls: types.CallData[],
+    ) => Promise<any>,
+  ) {
+    await promiseObjAll(
+      objMap(this.calls, async (chain, calls) => {
+        const connection = connectionFn(chain);
+        await executeFn(chain, connection, calls);
+      }),
+    );
+  }
+
+  protected connectionFn(chain: Chain) {
+    return this.checker.multiProvider.getChainConnection(chain);
+  }
+
+  protected ledgerConnectionFn(chain: Chain) {
+    const connection = this.checker.multiProvider.getChainConnection(chain);
+    return new ChainConnection({
+      signer: new LedgerSigner(connection.provider),
+      provider: connection.provider,
+      overrides: connection.overrides,
+      confirmations: connection.confirmations,
     });
   }
 
+  protected async estimateFn(
+    chain: Chain,
+    connection: ChainConnection,
+    calls: types.CallData[],
+  ) {
+    const signer = connection.signer;
+    if (!signer) throw new Error(`no signer found for ${chain}`);
+    const from = await signer.getAddress();
+    await Promise.all(
+      calls.map((call) =>
+        signer.estimateGas({
+          ...call,
+          from,
+        }),
+      ),
+    );
+  }
+
+  protected async sendFn(
+    chain: Chain,
+    connection: ChainConnection,
+    calls: types.CallData[],
+  ) {
+    const signer = connection.signer;
+    if (!signer) throw new Error(`no signer found for ${chain}`);
+    for (const call of calls) {
+      const response = await signer.sendTransaction(call);
+      console.log(`sent tx ${response.hash} to ${chain}`);
+      await response.wait(connection.confirmations);
+      console.log(`confirmed tx ${response.hash} on ${chain}`);
+    }
+  }
+
   estimateCalls() {
-    objMap(this.calls, async (chain, calls) => {
-      const connection = this.checker.multiProvider.getChainConnection(chain);
-      const signer = connection.signer;
-      if (!signer) {
-        throw new Error(`signer not found for ${chain}`);
-      }
-      for (const call of calls) {
-        await signer.estimateGas({ ...call, from: await signer.getAddress() });
-      }
-    });
+    return this.executeCalls(this.connectionFn, this.estimateFn);
```

Comment:
this is an unexpected naming to be used in `estimateCalls`
`executeCalls` seems like more of a `mapCalls`